cvat-core/src/annotations-collection.ts

```diff
@@ -1095,27 +1101,46 @@ export default class Collection {
         // eslint-disable-next-line no-unsanitized/method
         const imported = this.import(constructed);
         const importedArray = imported.tags.concat(imported.tracks).concat(imported.shapes);
+        const additionalUndo = [];
+        const additionalRedo = [];
+        const additionalClientIDs = [];
         for (const object of importedArray) {
-            if (object.shapeType === ShapeType.MASK && config.removeUnderlyingMaskPixels) {
-                (object as MaskShape).removeUnderlyingPixels(object.frame);
+            if (object.shapeType === ShapeType.MASK && config.removeUnderlyingMaskPixels.enabled) {
+                const {
+                    clientIDs,
+                    undo: undoWithRemoveIncorrectMasks,
+                    redo: redoWithRemoveIncorrectMasks,
+                } = (object as MaskShape).removeUnderlyingPixels(object.frame);
+                additionalUndo.push(undoWithRemoveIncorrectMasks);
+                additionalRedo.push(redoWithRemoveIncorrectMasks);
+                additionalClientIDs.push(clientIDs);
```

Comment:
Shall we be able to undo/redo any changes when removeUndrelyingPixels enabled?
I mean not only to restore deleted masks, but also cancel changes in all affected masks?

Right now that is what this feature lack of

Reply:
Added